### ISSUE TITLE
No new podman releases in ubuntu ppa

### ISSUE THREAD

**Issue by alitvak69** [containers/libpod#4213]
Dear developers,

Can you please push podman 1.6.1 and all dependencies to ubuntu ppa?  I thought it was happening automatically.

Thank you in advance,

**Comment by alitvak69** [comment]
Didn't mean to sound rude.  My apologies.
@lsm5 Does the failure on ppa has to do with the latest change in autobulder?

https://gitlab.com/lsm5/rpm-builder/commit/da3aa936697c5171da8ef5261caec4ba8e5bf5f4

Looks like credentials changed and it possibly broke the pipeline.  Just a thought.
`

**Comment by alitvak69** [comment]
```git
To gitlab:lsm5/podman.git
 + 8ec04502...3b589904 bionic -> bionic (forced update)
Branch 'bionic' set up to track remote branch 'bionic' from 'gitlab'.
Adding github mirror...
Pushing changes to github/podman...
To github:lsm5/podman.git
 + 1b423ce8...3b589904 bionic -> bionic (forced update)
Branch 'bionic' set up to track remote branch 'bionic' from 'github'.
Submitting build to PPA...
no such identity: /home/lsm5-bot/.ssh/id_rsa: No such file or directory
lsm5+bot@ppa.launchpad.net: Permission denied (publickey).
gpg: ../podman_1.6.1-1~ubuntu18.04~ppa1_source.changes: Error checking signature from 6FD26B91A57D528C: SignatureVerifyError: 0
gpg: ../podman_1.6.1-1~ubuntu18.04~ppa1.dsc: Error checking signature from 6FD26B91A57D528C: SignatureVerifyError: 0
Checking signature on .changes
Checking signature on .dsc
Uploading to ppa (via sftp to ppa.launchpad.net):
EOF during negotiation
E: Error connecting to remote host.
Failed to send build to PPA. Exiting...
ERROR: Job failed: exit code 1
```

**Comment by rfc1459** [comment]
Launchpad is rejecting the GPG key used to sign the changes and dsc uploads, there's also a build failure for buildah (the ubuntu-specific rebase is failing)

**Comment by lsm5** [comment]
ya, I probably messed up while changing identities. I'll put this on today's TODO.

**Comment by lsm5** [comment]
`.dput.cf` in the container image used for CI still had my older id, which was causing ssh issues. The pipeline at https://gitlab.com/lsm5/rpm-builder/pipelines/87610518 seems to be working. Failures in buildah are merge conflicts which I'll fix later today. HTH.

**Comment by rfc1459** [comment]
:tada: PPA is up-to-date (except for buildah):
```
# apt list --upgradable
Listing... Done
conmon/bionic 2.0.1-1~ubuntu18.04~ppa3 amd64 [upgradable from: 2.0.0-1~ubuntu18.04~ppa2]
containernetworking-plugins/bionic 0.8.2-1~ubuntu18.04~ppa3 amd64 [upgradable from: 0.8.2-1~ubuntu18.04~ppa1]
containers-common/bionic,bionic 0.1.39-1~ubuntu18.04~ppa3 all [upgradable from: 0.1.39-1~ubuntu18.04~ppa1]
containers-golang/bionic,bionic 0.3.1-1~ubuntu18.04~ppa3 all [upgradable from: 0.3.1-1~ubuntu18.04~ppa1]
containers-image/bionic,bionic 4.0.1-1~ubuntu18.04~ppa3 all [upgradable from: 3.0.2-1~ubuntu18.04~ppa1]
libvarlink/bionic 18-2~ubuntu18.04~ppa4 amd64 [upgradable from: 18-2~ubuntu18.04~ppa2]
podman/bionic 1.6.1-1~ubuntu18.04~ppa3 amd64 [upgradable from: 1.5.1-1~ubuntu18.04~ppa1]
skopeo/bionic 0.1.39-1~ubuntu18.04~ppa3 amd64 [upgradable from: 0.1.39-1~ubuntu18.04~ppa1]
slirp4netns/bionic 0.4.1-1~ubuntu18.04~ppa4 amd64 [upgradable from: 0.4.1-1~ubuntu18.04~ppa2]
```

**Comment by leorochael** [comment]
I guess this can be closed...

**Comment by shaicoleman** [comment]
Issue opened for buildah: https://github.com/containers/buildah/issues/1918

**Comment by shaicoleman** [comment]
Podman 1.6.3 not on Ubuntu not built

**Comment by alitvak69** [comment]
I thought that 1.6.3 was not released yet.

**Comment by rhatdan** [comment]
It is semi-released.  We have not built packages for Fedora yet.
@mheon What is the state?

**Comment by mheon** [comment]
It is released, but we're probably not building for Fedora. We're going to get a few more patches landed and make a separate 1.6.4 release specifically for Fedora (to better handle the F31 cgroupsv2 migration, among other things).

**Comment by alitvak69** [comment]
Nevertheless @lsm5 could you please check if the release can make it to ppa?  There are definitely some bug fixes I and others can be interested in.

**Comment by rfc1459** [comment]
Having 1.6.3 packaged for non-Fedora distros would finally land the fix for #4318, otherwise we'd have to wait for 1.6.5 (given that 1.6.4 would be a Fedora-specific release).

**Comment by rhatdan** [comment]
1.6.4 will not be a Fedora Specific release.  It is a release for anyone who wants to use cgroup V2.

**Comment by alitvak69** [comment]
So other than that is there a reason not to release package for ubuntu
users?  I believe the build on ppa is automatic.

**Comment by mheon** [comment]
I think it ought to be autobuilding.

**Comment by lsm5** [comment]
autobuilding yes, but afaict, it's broken atm, will take a look asap.

**Comment by alitvak69** [comment]
@lsm5 any news on 1.6.3 auto-building?

**Comment by alitvak69** [comment]
@lsm5 it looks like build of bionic failed with
`make[1]: Entering directory '/<<PKGBUILDDIR>>'
/usr/lib/go-1.13/bin/go generate ./cmd/podman/varlink/...
make[1]: /usr/lib/go-1.13/bin/go: Command not found
make[1]: *** [debian/rules:35: override_dh_auto_build] Error 127
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:24: build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
--------------------------------------------------------------------------------`

**Comment by lsm5** [comment]
> @lsm5 any news on 1.6.3 auto-building?

some upcoming change in setup: https://github.com/containers/libpod/issues/1742#issuecomment-555061785 . So please wait a little longer, apologies about the lag.

**Comment by alitvak69** [comment]
@lsm5 I wanted to get away from the flame war involving debian guys.  I wonder if can answer few questions here.  Sorry for nube ones.

In OBS, I see a successful build of 1.6.3~15 but when going to download repo link I don't see the package version there.  I see 1.6.3~14.  Is it normal?

How much of the work is needed to make ubuntu packaging somewhat complete?  If Debian guys care more about there distribution maybe you can put out ubuntu ones first.  It may be selfish but there is nobody doing work integrating podman and friends with ubuntu from ubuntu community.  We are currently rewriting our VoIP platform to get away from rkt and some other non-supported orchestration tools so we are following podman development very closely and packaging helps us to push ahead.  So at least we will test and report the bugs if we get ones.

**Comment by lsm5** [comment]
hey @alitvak69 appreciate your support, both podman use and on the thread! Not sure why the deb package isn't showing up, the buildlogs didn't show anything fishy. I just retriggered the build, let's see how it goes. And sure, I can prioritize ubuntu lts :)

**Comment by lsm5** [comment]
i can see the deb here https://build.opensuse.org/package/binaries/home:rhcontainerbot:stable/podman/xUbuntu_18.04 .. but there's another issue with containers-common installation, which I'm fixing now ..

**Comment by alitvak69** [comment]
@lsm5 
I see the package there as well
https://build.opensuse.org/package/binary/download/home:rhcontainerbot:stable/podman/xUbuntu_18.04/x86_64/podman_1.6.3~15_amd64.deb

However, it is not showing here:

https://download.opensuse.org/repositories/home:/rhcontainerbot:/stable/xUbuntu_18.04/

The download site is how I normally add and use repos in my opensuse and probably same for r other distros. The download repo link shows files dated by November 21st for bionic.  I.e. podman 1.6.3-14 and friends.

Looking forward to working with you and all libpod team members.

**Comment by lsm5** [comment]
> @lsm5
> I see the package there as well
> https://build.opensuse.org/package/binary/download/home:rhcontainerbot:stable/podman/xUbuntu_18.04/x86_64/podman_1.6.3~15_amd64.deb
> 
> However, it is not showing here:
> 
> https://download.opensuse.org/repositories/home:/rhcontainerbot:/stable/xUbuntu_18.04/

hmm, beats me. I'll check with OBS people on that.
 
> The download site is how I normally add and use repos in my opensuse and probably same for r other distros. The download repo link shows files dated by November 21st for bionic. I.e. podman 1.6.3-14 and friends.

Btw, do you mean you use the following to add repos (of course, podman isn't installable atm, but this should help for repo addition)

```
For xUbuntu 18.04 run the following:
Keep in mind that the owner of the key may distribute updates, packages and repositories that your system will trust (more information).

sudo sh -c "echo 'deb http://download.opensuse.org/repositories/home:/rhcontainerbot:/stable/xUbuntu_18.04/ /' > /etc/apt/sources.list.d/home:rhcontainerbot:stable.list"
wget -nv https://download.opensuse.org/repositories/home:rhcontainerbot:stable/xUbuntu_18.04/Release.key -O Release.key
sudo apt-key add - < Release.key
sudo apt-get update
sudo apt-get install podman
```

> 
> Looking forward to working with you and all libpod team members.

Likewise, thanks again! :)

**Comment by alitvak69** [comment]
Yes this is the link I use.  And if you go there and then click on amd64 you will not find latest podman package i.e. 1.6.3-15, just 1.6.3-14.

**Comment by lsm5** [comment]
> Yes this is the link I use. And if you go there and then click on amd64 you will not find latest podman package i.e. 1.6.3-15, just 1.6.3-14.

Ack yes, I meant if those were the steps you used to install packages, as the steps to enable it weren't obvious on the OBS homepage, but I see what you mean. I'll ping OBS people, but don't think I'll have anything before Monday.

**Comment by alitvak69** [comment]
@lsm5 Any news about 1.6.3-15 in the repo or / and if 1.6.3 becoming installable on bionic?

**Comment by lsm5** [comment]
hi @alitvak69 , sorry not yet, been involved in some other issues past 2 days, but today I managed to get skopeo and containers-common fixed, so the only pending thing is podman not being installable on 18.04, I'll try another rebuild later today, if it doesn't work, I'll ping OBS folks tomorrow morning.

**Comment by lsm5** [comment]
sent 1.6.4 to OBS just now, fingers crossed..

**Comment by lsm5** [comment]
hmm, seems like skopeo and containers-common are still not available on 18.04 repo (even though built successfully) along with podman. Let me check with OBS people tomorrow morning. Sorry about the delay.

**Comment by alitvak69** [comment]
Thank you for the updates.

**Comment by alitvak69** [comment]
@lsm5  Any news from OBS?  So far I see no changes.

**Comment by lsm5** [comment]
@alitvak69 no response from OBS people on IRC, I'll keep trying. Meanwhile, I've begun sending packages to the testing repo (mostly same as stable repo but will have rc content as well) as well to see if things are any different. 

The testing repo is at https://software.opensuse.org//download.html?project=home%3Arhcontainerbot%3Atesting&package=podman

Would you mind adding this repo as well and fetching podman from it? Please keep both stable and testing enabled since the latter doesn't have all packages included yet.

NOTE: This is only to get you unblocked immediately, and long term I plan to have everything working on a single repo itself.

**Comment by alitvak69** [comment]
Looks like their website down this morning.

**Comment by alitvak69** [comment]
I tested both repositories and were able to install podman 1.6.4 on ubuntu 18.04.   I hope that release propagation can be fixed.  They do have bugzilla FWIW

**Comment by lsm5** [comment]
hey @alitvak69  the testing repo is now self-contained and will have 1.7.0-rc1 in a bit (right now 1.6.4 is installable). Could you please give it a try and report back any issues?

**Comment by alitvak69** [comment]
Testing repo works self-contained, installed 1.7.0-rc1.  

It would be a good idea to fix stable because some people may stick with stable at some places.   

Any chance that buildah will make to those repos as well?

**Comment by lsm5** [comment]
Yes, I'll add buildah and fix stable too.

Btw, there's probably more changes upcoming to the repository location. I've been in discussions with @sysrich (thanks to @vrothberg) who maintains podman and other tools for openSUSE's kubic project, and I'll be moving our packages there, hopefully we can have a combined effort to accommodate majority distros and arches. I can update the stable subproject in there and get back to you.

Thanks again for helping out with testing!

**Comment by DaanDeMeyer** [comment]
Will the project-atomic PPA still be updated or do we have to use a new repo to get the latest podman release on Ubuntu? I'm not entirely sure if the the latest podman releases are already available via a package repository.

**Comment by lsm5** [comment]
PPA won't be updated anymore. All new stuff going to OBS.

The `kubic:libcontainers:stable` project repo works now for podman 1.6.4, I'm working on buildah and cri-o now. Repo setup instructions can be found here: https://software.opensuse.org//download.html?project=devel%3Akubic%3Alibcontainers%3Astable&package=podman 

Let me know if any issues..

**Comment by lsm5** [comment]
buildah 1.12.0 should be available as well, cri-o coming up..

**Comment by b0wter** [comment]
Works perfectly for me. Thank you! 
It would be great if someone updated the installation instructions on the official homepage:

https://podman.io/getting-started/installation

**Comment by lsm5** [comment]
> Works perfectly for me. Thank you!
> It would be great if someone updated the installation instructions on the official homepage:
> 
> https://podman.io/getting-started/installation

yup, I'll send a PR today. Thanks for testing this!

**Comment by alitvak69** [comment]
@lsm5 Found the issue with 1.6.4 build today, please take a look at #4734 
Thank you

**Comment by alitvak69** [comment]
@lsm5 Sorry for bugging you, but could you add seccomp to your tags?

**Comment by lsm5** [comment]
> @lsm5 Sorry for bugging you, but could you add seccomp to your tags and push a new build?

No worries, always a pleasure to help dedicated podman users :) .. I sent 1.6.4\~2 to OBS stable repo just now, should be available in an hour or less I guess.

**Comment by Knapoc** [comment]
is there a reason why conmon is not provided in the repo for arm64?

**Comment by lsm5** [comment]
> is there a reason why conmon is not provided in the repo for arm64?

@Knapoc the OBS build env is messed up for ubuntu on non-x86_64 arches. I'll nag them about it after new years. Happy holidays!
